app_dart/lib/src/request_handlers/postsubmit_luci_subscription_v2.dart

```diff
@@ -0,0 +1,174 @@
+    log.info('Found $firestoreTask');
+
+    if (_shouldUpdateTask(build, firestoreTask)) {
+      final String oldTaskStatus = firestoreTask.status;
+      firestoreTask.updateFromBuildV2(build);
+
+      log.info('Updated firestore task $firestoreTask');
+
+      task.updateFromBuildbucketV2Build(build);
+      await datastore.insert(<Task>[task]);
+      final List<Write> writes = documentsToWrites([firestoreTask], exists: true);
+      await firestoreService.batchWriteDocuments(BatchWriteRequest(writes: writes), kDatabase);
+      log.fine('Updated datastore from $oldTaskStatus to ${firestoreTask.status}');
+    } else {
+      log.fine('skip processing for build with status scheduled or task with status finished.');
+    }
+
+    final Commit commit = await datastore.lookupByValue<Commit>(commitKey);
+    final CiYaml ciYaml = await scheduler.getCiYaml(commit);
+    final List<Target> postsubmitTargets = ciYaml.postsubmitTargets;
+    if (!postsubmitTargets.any((element) => element.value.name == firestoreTask!.taskName)) {
+      log.warning('Target ${firestoreTask.taskName} has been deleted from TOT. Skip updating.');
+      return Body.empty;
+    }
+    final Target target =
+        postsubmitTargets.singleWhere((Target target) => target.value.name == firestoreTask!.taskName);
+    if (firestoreTask.status == firestore.Task.statusFailed ||
+        firestoreTask.status == firestore.Task.statusInfraFailure ||
+        firestoreTask.status == firestore.Task.statusCancelled) {
+      log.fine('Trying to auto-retry...');
+      final bool retried = await scheduler.luciBuildService.checkRerunBuilder(
+        commit: commit,
+        target: target,
+        task: task,
+        datastore: datastore,
+        taskDocument: firestoreTask,
+        firestoreService: firestoreService,
+      );
+      log.info('Retried: $retried');
+    }
+
+    // Only update GitHub checks if target is not bringup
+    if (target.value.bringup == false && config.postsubmitSupportedRepos.contains(target.slug)) {
+      log.info('Updating check status for ${target.getTestName}');
+      await githubChecksService.updateCheckStatus(
+        build: build,
+        userDataMap: userDataMap,
+        luciBuildService: scheduler.luciBuildService,
+        slug: commit.slug,
+      );
+    }
+
+    return Body.empty;
+  }
+
+  // No need to update task in datastore if
+  // 1) the build is `scheduled`. Task is marked as `In Progress`
+  //    whenever scheduled, either from scheduler/backfiller/rerun. We need to update
+  //    task in datastore only for
+  //    a) `started`: update info like builder number.
+  //    b) `completed`: update info like status.
+  // 2) the task is already completed.
+  //    The task may have been marked as completed from test framework via update-task-status API.
+  bool _shouldUpdateTask(bbv2.Build build, firestore.Task task) {
+    return build.status != bbv2.Status.SCHEDULED && !Task.finishedStatusValues.contains(task.status);
```

Comment:
```suggestion
    return build.status != bbv2.Status.SCHEDULED && !firestore.Task.finishedStatusValues.contains(task.status);
```

Reply:
Diff is stale but I have added the change. Good catch.